### ISSUE TITLE
After Pulling Still Warned "Local changes detected. Pulling changes from the cloud will override your local changes."

### ISSUE THREAD

**Issue by pnewhook** [aws-amplify/amplify-cli#6373]
**Describe the bug**
No matter how many times I run `amplify pull` I'm still warned about potentially overwriting local changes.

```
PS C:\Users\peter\dev\track-titan-web> amplify pull
Initializing new Amplify CLI version...
Done initializing new version.
Scanning for plugins...
Plugin scan successful

Pre-pull status:

Current Environment: dev

| Category | Resource name         | Operation | Provider plugin   |
| Storage  | Poseidon              | No Change | awscloudformation |
| Hosting  | amplifyhosting        | No Change | awscloudformation |
| Auth     | tracktitanwebe5df7ee5 | No Change | awscloudformation |
| Function | webapifunction        | No Change | awscloudformation |
| Api      | webapi                | No Change | awscloudformation |

Local changes detected.
Pulling changes from the cloud will override your local changes.
? Are you sure you would like to continue? Yes
√ Successfully pulled backend environment dev from the cloud.

Post-pull status:

Current Environment: dev

| Category | Resource name         | Operation | Provider plugin   |
| -------- | --------------------- | --------- | ----------------- |
| Storage  | Poseidon              | No Change | awscloudformation |
| Hosting  | amplifyhosting        | No Change | awscloudformation |
| Auth     | tracktitanwebe5df7ee5 | No Change | awscloudformation |
| Function | webapifunction        | No Change | awscloudformation |
| Api      | webapi                | No Change | awscloudformation |

PS C:\Users\peter\dev\track-titan-web> amplify pull

Pre-pull status:

Current Environment: dev

| Category | Resource name         | Operation | Provider plugin   |
| -------- | --------------------- | --------- | ----------------- |
| Storage  | Poseidon              | No Change | awscloudformation |
| Hosting  | amplifyhosting        | No Change | awscloudformation |
| Auth     | tracktitanwebe5df7ee5 | No Change | awscloudformation |
| Function | webapifunction        | No Change | awscloudformation |
| Api      | webapi                | No Change | awscloudformation |

Local changes detected.
Pulling changes from the cloud will override your local changes.
? Are you sure you would like to continue? (y/N)
```
**Amplify CLI Version**
4.41.1

**Expected behavior**
After pulling, should not be warned about overwriting local changes.

**Desktop (please complete the following information):**
 - Windows
 - v14.15.3

**Comment by attilah** [comment]
@pnewhook if you have your files under git and you start the pull from a clean slate, is there any files that will be detected by git as changed after the pull? That could help us diagnose what could be the problem in your case.

Can you provider more information on what type of hosting you have and what language runtime your function is created with?

**Comment by pnewhook** [comment]
@attilah After cloning my project into a new directory I ran `amplify init` (I hope that's the correct command, documentation on initializing amplify for existing projects is sparse). That modified a handful of files in my local directory
![2021-01-13 19_25_25-Sourcetree](https://user-images.githubusercontent.com/92876/104528646-d4b5a080-55d5-11eb-914e-439820e7f5e4.png)
Most of the files are unchanged which I suspect means there were line ending changes. A couple of the files were genuinely modified
![2021-01-13 19_25_31-Sourcetree](https://user-images.githubusercontent.com/92876/104528690-ef881500-55d5-11eb-8fd6-2661360f470d.png)
![2021-01-13 19_25_38-Sourcetree](https://user-images.githubusercontent.com/92876/104528695-f151d880-55d5-11eb-9311-d4a863b9af21.png)
 I then ran `amplify pull` That caused many more files to appear updated without content changes, again, I suspect line ending issues. I'm warned that I will overwrite local changes, despite having just pulled
![2021-01-13 19_26_47-Windows PowerShell](https://user-images.githubusercontent.com/92876/104528785-2f4efc80-55d6-11eb-90ab-7af68e7f88d7.png)

![2021-01-13 19_28_10-Sourcetree](https://user-images.githubusercontent.com/92876/104528901-658c7c00-55d6-11eb-953e-e1a46f0486d2.png)

I'm using Hosting with Amplify Console. I have a react web application with a node.js function.

**Comment by pnewhook** [comment]
All of the modified files are changed from CRLF in my working directory to LF after `amplify pull/init`

**Comment by attilah** [comment]
As the CLI is using hashing and it seems that your git config is set to auto translate between line endings it can cause the CLI to see differences as the hashes of files/directories will be different.

Closing the issue as it is a non-issue, but local configuration related side-effect. If you think that something else is happening feel free to comment on the issue to reopen it.

**Comment by pnewhook** [comment]
I agree it's a local configuration issue but I'm not sure how to deal with it. Letting git automatically convert CRLF to LF on commit, and vice versa on checkout is very common on Windows. Does the CLI need the repo to use LF line endings?

**Comment by pnewhook** [comment]
I changed every file in my repo to use LF endings, then pushed, but `amplify pull` still warning me that local changes will be overridden. Except now nothing changes in git after pulling.

**Comment by pnewhook** [comment]
I tried including all the files that were ignored under /amplify and the only file that's changing after amplify push/pull/pull is the amplify/backend/amplify-meta.json with updates to `lastPushTimeStamp`

**Comment by attilah** [comment]
when you change the settings and convert to LFs the first time it is OK to detect differences, since the cloud uploaded Amplify files ARE different. `amplify push` always does a `pull` to make sure changes pushed by others will not be overwritten so if you do the LF conversion, then proceed with the push ignoring the changes, then the new LF endings will be uploaded and subsequent pushes should not detect further changes.

**Comment by pnewhook** [comment]
Yes, I fully expect to see change warning after manually switching to LF, then I pushed. But no matter how many time I `amplify pull` I keep getting the warning that I have local changes that will be overwritten. Are there any debug logs that show exactly what amplify thinks is different?

**Comment by JvdWest** [comment]
I struggled with the whole LF/CLRF issue with Amplify on Windows. I would do an Amplify pull and then a bunch of files would have "changes", but actually just LF/CRLF changes, and if you Git revert those guys, then you would have Amplify changes. I did change settings in my IDE, but it was resetting the whole time when I do pull/push. One thing that worked for me was to remove all Git changes (e.g. git stash). Then this:

`git config core.autocrlf input`

Then amplify pull again. Then there were no Git changes, and all my Amplify resources were up to date.
